ddtrace/ddtrace.stub.php

```diff
@@ -0,0 +1,1051 @@
+     * - 'prehook': a closure to be run prior to the function call
+     * - 'posthook': a closure to be run after the function was executed
+     * - 'instrument_when_limited': set to 1 shall the function be traced in limited mode (e.g., when span limit
+     * exceeded)
+     * - 'recurse': a boolean to state whether should recursive calls be traced as well
+     * @param \Closure(object $object, string $scope, array $args, mixed $retval, \Exception|null $exception):void|null $posthook A post-hook function that will be called after
+     * the instrumented method is executed. This can be useful for things like formatting output data or logging the
+     * results of the method call.
+     *
+     * Note that the $posthook argument shouldn't be given when calling hook_function if a configuration array has been
+     * passed.
+     * @return bool 'false' when no hook is passed, else 'true'
+     */
+    function hook_method(
+        string $className,
+        string $methodName,
+        \Closure|array|null $prehookOrConfigArray = null,
+        ?\Closure $posthook = null
+    ): bool {}
+
+    // phpcs:enable Generic.Files.LineLength.TooLong
+
+    /**
+     * Add a tag to be automatically applied to every span that is created, if tracing is enabled.
+     *
+     * @param string $key Tag key
+     * @param string $value Tag Value
+     */
+    function add_global_tag(string $key, string $value): void {}
+
+    /**
+     * Add a tag to be propagated along distributed traces' information. It also adds the tag to the local root span.
+     *
+     * @param string $key Tag key
+     * @param string $value Tag value
+     */
+    function add_distributed_tag(string $key, string $value): void {}
+
+    /**
+     * Add user information to monitor authenticated requests in the application.
+     *
+     * @param string $userId Unique identifier of the user (usr.id)
+     * @param array $metadata User monitoring tags (usr.<TAG_NAME>) applied to the 'meta' section of the root span
+     * @param bool|null $propagate If set to 'true', user's information will be propagated in distributed traces
+     */
+    function set_user(string $userId, array $metadata = [], bool|null $propagate = null): void {}
+
+    /**
+     * Close child spans of a parent span if a non-internal span is given,
+     * else if 'null' is given, close active, non-internal spans
+     *
+     * @param SpanData|null $span The parent span
+     * @return false|int 'false' if spans couldn't be closed, else the number of span closed
+     */
+    function close_spans_until(?SpanData $span): false|int {}
+
+    /**
+     * Get the active span
+     *
+     * @return SpanData|null 'null' if tracing isn't enabled or there isn't any active span, else the active span
+     */
+    function active_span(): null|SpanData {}
+
+    /**
+     * Get the root span
+     *
+     * @return SpanData|null 'null' if tracing isn't enabled or if the active stack doesn't have a root span,
```

Comment:
Return type in PHPDoc differs from signature. Could you please update type hints?